backend/src/modules/location/location_service.py

```
@@ -0,0 +1,47 @@
+from sqlalchemy import delete, update
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.future import select
+from src.modules.location.location_entity import LocationEntity
+from src.modules.location.location_model import LocationCreate, LocationUpdate
+
+
+class LocationService:
+    def __init__(self, session: AsyncSession):
+        self.session = session
+
+    async def get_all_locations(self):
+        """Return all locations"""
+        result = await self.session.execute(select(LocationEntity))
+        return result.scalars().all()
+
+    async def get_location_by_id(self, location_id: int):
+        """Get a single location by ID"""
+        result = await self.session.execute(
+            select(LocationEntity).where(LocationEntity.id == location_id)
+        )
+        return result.scalar_one_or_none()
```

Take a look at `user_service.py`. Instead of returning None for a not found, throw a custom `LocationNotFoundException`